plugins/connections/ui/src/main/java/org/pentaho/di/connections/ui/tree/ConnectionFolderProvider.java

```diff
@@ -58,4 +60,14 @@ public void refresh( AbstractMeta meta, TreeNode treeNode, String filter ) {
   public String getTitle() {
     return STRING_VFS_CONNECTIONS;
   }
+
+  @Override
+  public void create( AbstractMeta meta, TreeNode parent ) {
+    Repository repository = Spoon.getInstance().getRepository();
+    if( repository != null && repository.getUserInfo() != null && !repository.getUserInfo().isAdmin() ) {
```

Comment:
Add unit test for ConnectionFolderProvider.java#create. 

It should cover 

- when a user is not connected to a repository
- when a user is connected to a repository and is not Admin
- when a user is connected to a repository and isAdmin

Reply:
Added Unit tests